// hexo-theme-icarus/layout/common/scripts.jsx
const { Component, Fragment } = require('inferno');
const Plugins = require('./plugins');

module.exports = class extends Component {
    render() {
        const { site, config, helper, page } = this.props;
        const { url_for, cdn } = helper;
        const { article } = config;
        const language = page.lang || page.language || config.language || 'en';
        
		const busuanzi_init = `if (${config.plugins.busuanzi} == true) {
			$(document).ready(function () {
				var int = setInterval(fixCount, 100);
				var uvOffSet = parseInt(${config.busuanzi.site_uv_offset | 0});
				var pvOffSet = parseInt(${config.busuanzi.site_pv_offset | 0});
				
				function fixCount() {
					var realUv = parseInt($("#busuanzi_value_site_uv").html())
					var realPv = parseInt($("#busuanzi_value_site_pv").html())
					if ($("#busuanzi_container_site_uv").css("display") != "none" && realUv > 0) {
						clearInterval(int);
						$("#busuanzi_value_site_uv").html(realUv + uvOffSet);
						$("#busuanzi_value_site_pv").html(realPv + pvOffSet);
					}
				}
			});
		}`;


        let fold = 'unfolded';
        let clipboard = true;
        if (article && article.highlight) {
            if (typeof article.highlight.clipboard !== 'undefined') {
                clipboard = !!article.highlight.clipboard;
            }
            if (typeof article.highlight.fold === 'string') {
                fold = article.highlight.fold;
            }
        }

        const embeddedConfig = `var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: ${clipboard},
                    fold: '${fold}'
                }
            }
        };`;

        return <Fragment>
            <script src={cdn('jquery', '3.3.1', 'dist/jquery.min.js')}></script>
            <script src={cdn('moment', '2.22.2', 'min/moment-with-locales.min.js')}></script>
            {clipboard && <script src={cdn('clipboard', '2.0.4', 'dist/clipboard.min.js')} defer></script>}
            <script dangerouslySetInnerHTML={{ __html: `moment.locale("${language}");` }}></script>
            <script dangerouslySetInnerHTML={{ __html: embeddedConfig }}></script>
            <script dangerouslySetInnerHTML={{ __html: busuanzi_init }}></script>
            <script src={url_for('/js/column.js')}></script>
            <Plugins site={site} config={config} page={page} helper={helper} head={false} />
            <script src={url_for('/js/main.js')} defer></script>

            {/* <!-- 雪花特效 -->
            <script type="text/javascript">
            var windowWidth = $(window).width();
            if (windowWidth > 480) {
                document.write('<script type="text/javascript" src={url_for('/js/src/snow.js')}><\/script>');
            }
            </script> */}
            {/* <script src={url_for('/js/src/snow.js')} type="text/javascript"></script> */}


        </Fragment>;
    }
};
